packages/fxa-auth-server/lib/routes/account.ts

```diff
@@ -502,6 +504,54 @@ export class AccountHandler {
     });
   }
 
+  async accountStub(request: AuthRequest) {
+    this.log.begin('Account.stub', request);
+    const { email, clientId } = request.payload as any;
+    await this.customs.check(request, email, 'accountCreate');
+
+    if (this.OAUTH_DISABLE_NEW_CONNECTIONS_FOR_CLIENTS.has(clientId)) {
+      throw error.disabledClientId(clientId);
+    }
+
+    const client = await getClientById(clientId);
+
+    await this.deleteAccountIfUnverified(request, email);
```

Comment:
This leaves a small window where the account could be deleted while the user is in the process of subscribing. If it were to happen the most common result would be adding the subscription would fail but there may be an **extremely** small window where the sub could be added and then account deleted. I don't think this will happen in real world use though, so I'm not very concerned about making it race-safe.

Reply:
Seems reasonable to me, I don't think this will be a common case either.